bin/src/main.rs

```diff
@@ -67,22 +66,32 @@ fn main() {
             Err(e) => warn!("{}", e),
         };
     }
+    executor.init();
 
-    let mut source_reader = SourceReader::new();
-    register_journald_source(&mut source_reader);
-    source_reader.register(FSSource::new(config.log.dirs, config.log.rules));
+    let mut fs_tailer_buf = [0u8; 4096];
```

Comment:
Yeh, having the caller provide stratch space for the callee is a little ugly, it makes things simpler when things need to live long enough to get back to the callee and you'd prefer not to allocate, but it can sometimes be worked around. I'll see if I can smuggle the buffer around using a once_cell or an additional wrapper object, but we might be stuck with this...